test/e2e/clustergroup_test.go

```diff
@@ -20,28 +20,108 @@ import (
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	corev1a1 "github.com/vmware-tanzu/antrea/pkg/apis/core/v1alpha2"
 	secv1alpha1 "github.com/vmware-tanzu/antrea/pkg/apis/security/v1alpha1"
 )
 
 func testInvalidCGIPBlockWithPodSelector(t *testing.T) {
-	invalidErr := fmt.Errorf("clustergroup created with ipblock and pod selector")
+	invalidErr := fmt.Errorf("clustergroup created with ipblock and podSelector")
 	cgName := "ipb-pod"
 	pSel := &metav1.LabelSelector{MatchLabels: map[string]string{"pod": "x"}}
 	cidr := "10.0.0.10/32"
 	ipb := &secv1alpha1.IPBlock{CIDR: cidr}
-	if _, err := k8sUtils.CreateOrUpdateCG(cgName, pSel, nil, ipb); err == nil {
+	cg := &corev1a1.ClusterGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cgName,
+		},
+		Spec: corev1a1.GroupSpec{
+			PodSelector: pSel,
+			IPBlock:     ipb,
+		},
+	}
+	if _, err := k8sUtils.CreateOrUpdateCG(cg); err == nil {
 		// Above creation of CG must fail as it is an invalid spec.
 		failOnError(invalidErr, t)
 	}
 }
 
 func testInvalidCGIPBlockWithNSSelector(t *testing.T) {
-	invalidErr := fmt.Errorf("clustergroup created with ipblock and namespace selector")
+	invalidErr := fmt.Errorf("clustergroup created with ipblock and namespaceSelector")
 	cgName := "ipb-ns"
-	nSel := &metav1.LabelSelector{MatchLabels: map[string]string{"pod": "x"}}
+	nSel := &metav1.LabelSelector{MatchLabels: map[string]string{"ns": "y"}}
 	cidr := "10.0.0.10/32"
 	ipb := &secv1alpha1.IPBlock{CIDR: cidr}
-	if _, err := k8sUtils.CreateOrUpdateCG(cgName, nil, nSel, ipb); err == nil {
+	cg := &corev1a1.ClusterGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cgName,
+		},
+		Spec: corev1a1.GroupSpec{
+			NamespaceSelector: nSel,
+			IPBlock:           ipb,
+		},
+	}
+	if _, err := k8sUtils.CreateOrUpdateCG(cg); err == nil {
+		// Above creation of CG must fail as it is an invalid spec.
+		failOnError(invalidErr, t)
+	}
+}
+
+func testInvalidCGServiceRefWithPodNSSelector(t *testing.T) {
```

Comment:
I see no reason for not splitting this test into 2 parts (podSelector / namespaceSelector)?

Reply:
any thoughts?

Reply:
Sorry should comment on this. I changed this locally just has not pushed awaiting your comments on the other stuff...